server/utils/courseParticipationUtils.ts

```diff
@@ -239,10 +233,11 @@ export default class CourseParticipationUtils {
 
     if (hasError) {
       request.flash('formValues', JSON.stringify(request.body))
-      return { hasError, value: undefined }
     }
 
-    return { hasError, value: numericValue }
+    const processedValue = !trimmedValue || hasError ? undefined : numericValue
```

Comment:
I'm finding this a bit difficult to parse, I wonder if this might be DRYness at the expense of readability - was it causing headaches before?

Reply:
Not really, it was just three lines doing similar things. Could change it to something like this:

```ts
const shouldReturnUndefinedValue = !trimmedValue || hasError
const processedValue = shouldReturnUndefinedValue ? undefined : numericValue
```

Happy to leave it though if the more verbose version is easier to understand - there's no performance gain here

Reply:
I personally prefer the more readable approach over the DRY one, could we leave as-is?

Reply:
Yep, no problem!